### ISSUE TITLE
`assert-must-preload` compatible with `loadAll()`?

### ISSUE THREAD

**Issue by kumy** [embermap/ember-data-storefront#66]
I'm wondering if `assert-must-preload` component is compatible with the `loadAll()` function.

I have a component template such as:
```
// templates/components/geokret-details.hbs

{{assert-must-preload geokret "owner"}}
<h1>{{geokret.name}}</h1>
Owner: {{geokret.owner.name}}
```

Route template for details:
```
// templates/geokrety/details.hbs

{{geokret-details geokret=model}}
```

My route for visualizing just one item is as follow:
```
// routes/geokrety/details.js

import Route from '@ember/routing/route';
export default Route.extend({
  model(params) {
    return this.store.loadRecord('geokret', params.geokret_id, {
      include: 'owner'
    });
  }
});
```
If I wish to display one GeoKret details it's working fine, but if I wish to display a list of GeoKrety, it fails with error `Error: Assertion Failed: You tried to render a component:geokret-details that accesses relationships off of a geokret, but that model didn't have all of its required relationships preloaded ('owner'). Please make sure to preload the association. [ember-data-storefront]`

Route template for list:
```
// templates/geokrety/index.hbs

{{#each model as |geokret|}}
  {{geokret-details geokret=geokret}}
{{/each}}
```

```
// routes/geokrety/index.js

import Route from '@ember/routing/route';
export default Route.extend({
  model() {
    return this.store.loadAll('geokret', {
      page: {
        size: 2,
        number: 1
      },
      include: 'owner'
    });
  }
});
```

---

```
Ember           : 3.7.3
Ember Data      : 3.7.0
jQuery          : 3.3.1
Ember Bootstrap : 2.4.0
```
Does someone see what I'm doing wrong?

I've pushed [all my code](https://github.com/geokrety/geokrety-frontend/tree/fail/app) there in case someone need to look deeper.

Thanks!

**Comment by ryanto** [comment]
Thanks! This is a bug. Right now `loadAll` doesn't pass the relationships that were sideloaded to the models created by Ember Data.

As a way to get you unstuck today, I'd drop `assert-must-preload` and inside the `geokret-details` `didInsertElement` hook use the references API to check if the relationship was loaded.

```
// geokret-details

didInsertElement() {
  if (this.geokret.belongsTo('owner').value()) {
    // the geokret's owner wasn't loaded (or it's null). 
    throw "load the owner";
  }
}
```

The only downside to this approach is that it will fail when the owner is null. If you run into that let me know and I can give you some pointers.

**Comment by kumy** [comment]
@ryanto thanks. I have to test that tonight. Regarding the `null` case, I'll encounter the problem for the 'holder' field. I would love to see how you solve that. Thanks again.

**Comment by ryanto** [comment]
Hey @kumy, this fell off my radar, but it should be fixed now. It's merged into master and released in `0.16.1`, see release here: https://github.com/embermap/ember-data-storefront/releases/tag/v0.16.1

Let me know how it goes, happy to make more changes to our relationship tracking based on your feedback!
